lib/Scrabble.rb

```diff
@@ -0,0 +1,94 @@
+module Scrabble
+
+  # constant scoring for scrabble
+ LETTERS = {
+   1 => ["A", "E", "I", "O", "U", "L", "N", "R", "S", "T"],
+   2=> ["D", "G"],
+   3=> ["B", "C", "M", "P"],
+   4=> ["F", "H", "V", "W", "Y"],
+   5=> "K",
+   8=> ["J", "X"],
+   10=> ["Q", "Z"]
+ }
+
+  class Scrabble
+
+    def self.is_valid?(input_letters)
+    # confirming that input is valid, only accepting letters
+
+      if input_letters == []
+        return "please type something in."
+        exit
+      end
+
+      possible_letters = %w(Q W E R T Y U I O P A S D F G H J K L Z X C V B N M)
+
+      input_letters.each do |letter|
+
+        if possible_letters.include?(letter)
+          return input_letters
+        else
+          return "Error, not valid letters."
+          exit
+        end
+      end
+
+    end
+
+    def self.score(word)
+      total_points = 0
+      word.upcase!
+      input_letters = letter_array(word) # array of letters
+      # calling the validating method is_valid?(word)
+      is_valid?(input_letters)
+      points_array = letter_scoring(input_letters)
+
+      points_array.each do |point_value|
+        total_points += point_value
+      end
+
+      return total_points
+    end
+
+    def self.letter_scoring(input_letters) # input_letters is an array of letters
+      points_array = []
+
+      input_letters.each do |letter|
```

Comment:
It seems like you could potentially write this in a more reusable way - any ideas how to do that?